cmd/proxy-server/main_test.go

```diff
@@ -109,3 +130,391 @@ func TestUpdateStorageSystems(t *testing.T) {
+
+			getRolesFn := func(ctx context.Context) (*roles.JSON, error) {
+				return &rff, nil
+			}
+			svc := role.NewService(fakeKube{GetConfiguredRolesFn: getRolesFn}, successfulValidator{})
+
+			//create volume
+			rdb.HSetNX("quota:powerflex:542a2d5f5122210f:bronze:PancakeGroup-3:data", "vol:k8s-6aac50817e:capacity", 1)
+			rdb.HSetNX("quota:powerflex:542a2d5f5122210f:steel:PancakeGroup-3:data", "vol:k8s-6aac50818e:capacity", 1)
+
+			//list volumes test
+
+			h := volumesHandler(&roleClientService{roleService: svc}, rdb, jwx.NewTokenManager(jwx.HS256), log)
+			w := httptest.NewRecorder()
+			r, err := http.NewRequestWithContext(ctx, http.MethodGet, "/proxy/volumes/", nil)
+			r.Header.Add("Authorization", "Bearer "+string(decAccTkn))
+
+			checkError(t, err)
+
+			h.ServeHTTP(w, r)
+
+			//check if endpoint returns OK status
+			if got := w.Result().StatusCode; got != http.StatusOK {
+				t.Errorf("got %d, want %d", got, http.StatusOK)
+			}
+			return
+		},
+		"Successfull run of deleted Role": func(t *testing.T, ctx context.Context, rdb *redis.Client, log *logrus.Entry) {
+			//creates tenant and binds role by name
+			name := "PancakeGroup-4"
+			createTenant(t, sut, tenantConfig{Name: name, Roles: "CA-medium-4,CA-large-4"})
+
+			tkn, err := sut.GenerateToken(context.Background(), &pb.GenerateTokenRequest{
+				TenantName: name,
+			})
+
+			tknData := tkn.Token
+			var tokenData struct {
+				Data struct {
+					Access string `yaml:"access"`
+				} `yaml:"data"`
+			}
+			err = yaml.Unmarshal([]byte(tknData), &tokenData)
+			checkError(t, err)
+			decAccTkn, err := base64.StdEncoding.DecodeString(tokenData.Data.Access)
+			checkError(t, err)
+			//create Roles
+			roleInstance, err := roles.NewInstance("CA-medium-4", "powerflex", "542a2d5f5122210f", "bronze", "9GB")
+			roleInstanceTwo, err := roles.NewInstance("CA-large-4", "powerflex", "542a2d5f5122210f", "bronze", "20GB")
+			checkError(t, err)
+
+			rff := roles.NewJSON()
+
+			err = rff.Add(roleInstance)
+			checkError(t, err)
+
+			err = rff.Add(roleInstanceTwo)
+			checkError(t, err)
+
+			getRolesFn := func(ctx context.Context) (*roles.JSON, error) {
+				return &rff, nil
+			}
+			svc := role.NewService(fakeKube{GetConfiguredRolesFn: getRolesFn}, successfulValidator{})
+
+			//create volume
+			rdb.HSetNX("quota:powerflex:542a2d5f5122210f:bronze:PancakeGroup-4:data", "vol:k8s-6aac50817e:capacity", 1)
+			rdb.HSetNX("quota:powerflex:542a2d5f5122210f:bronze:PancakeGroup-4:data", "vol:k8s-6aac50818e:deleted", 1)
+
+			//list volumes test
+
+			h := volumesHandler(&roleClientService{roleService: svc}, rdb, jwx.NewTokenManager(jwx.HS256), log)
+			w := httptest.NewRecorder()
+			r, err := http.NewRequestWithContext(ctx, http.MethodGet, "/proxy/volumes/", nil)
+			r.Header.Add("Authorization", "Bearer "+string(decAccTkn))
+
+			checkError(t, err)
+
+			h.ServeHTTP(w, r)
+
+			//check if endpoint returns OK status
+			if got := w.Result().StatusCode; got != http.StatusOK {
```

Comment:
Is checking only the status code good enough for the test? I think we should verify the list of volumes in the response too.

Reply:
At this point this is rolling target, final return value will changed after storage-service integration, for now I think it is fine.